src/Bucket.hpp

```diff
@@ -53,10 +53,25 @@ class Bucket {
     }
 
     bool is_full() {
-        return false;
+      char page_data[PAGE_SIZE];
+      dm->read_page(page_id, page_data);
+      std::istringstream ss(std::stringstream::in | std::stringstream::binary);
+      char* end;
+      uint64_t size = strtoull(page_data, &end, 10);
+      if(size==PAGE_SIZE)
```

Comment:
`size` is only the size of the unordered_map. We also need to account for the bytes taken up by `size` variable itself. Hint: the `end` variable points to the space character(` `) after the size.
```
123 SomeGibberishHere....
   ^end
```